internal/provider/cluster_resource.go

```diff
@@ -1605,3 +1733,41 @@ func reconcileRegionUpdate(
 func NewClusterResource() resource.Resource {
 	return &clusterResource{}
 }
+
+func byocEqual(a *CustomerCloudAccount, b *CustomerCloudAccount) bool {
```

Comment:
It would be nice to have some tests for this.

Reply:
Okay I added some tests.